pyam_analysis/core.py

```diff
@@ -803,26 +819,40 @@ def read_data(fname, regions=None, *args, **kwargs):
     else:
         df = pd.read_excel(fname, *args, **kwargs)
 
-    df = df.rename(columns={c: str(c).lower() for c in df.columns})
+    return(format_data(df))
+
+
+def format_data(data, regions=None):
```

Comment:
suggest taking out regions argument, this should be filtered by the user later if desired.